pkg/issuer/acme/dns/route53/route53.go

```diff
@@ -91,101 +109,38 @@ func (d *sessionProvider) GetSession(ctx context.Context) (aws.Config, error) {
 		d.log.V(logf.DebugLevel).Info("not using ambient credentials")
 		optFns = append(optFns, config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(d.AccessKeyID, d.SecretAccessKey, "")))
 	}
-
 	cfg, err := config.LoadDefaultConfig(ctx, optFns...)
 	if err != nil {
 		return aws.Config{}, fmt.Errorf("unable to create aws config: %s", err)
 	}
-
-	// For backwards compatibility with cert-manager <= 1.14, where we used the aws-sdk-go v1
-	// library, we configure the SDK here to use the global sts endpoint. This was the default
-	// behaviour of the SDK v1 library, but has to be explicitly set in the v2 library. For the
-	// route53 calls, we use the region provided by the user (see below).
-	stsCfg := cfg.Copy()
-	stsCfg.Region = "aws-global"
```

Comment:
I've changed my mind about this.
I don't think backwards compatibility is a good reason for hard coding the STS region to `aws-global`.
I think users will want cert-manager to use the STS endpoint that is chosen by the AWS-SDK and to be able to use all the mechanisms that it provides for configuring the STS endpoints, such as `AWS_STS_REGIONAL_ENDPOINTS`: 
 * https://docs.aws.amazon.com/sdkref/latest/guide/feature-sts-regionalized-endpoints.html

And `AWS_ENDPOINT_URL_STS`:
 * https://docs.aws.amazon.com/sdkref/latest/guide/feature-ss-endpoints.html#ss-endpoints-envar

I found one possible justification for being able to override the STS endpoint settings and a proposed AWS-SDK v2 feature to make it possible, here:
 * https://github.com/aws/aws-sdk-go-v2/issues/2686#issuecomment-2221094539

Reply:
We discussed this further in the cert-manager standup this morning and decided that there are good reasons **not** to use the global STS endpoints, summarized in this blog post:
 * https://aws.amazon.com/blogs/security/how-to-use-regional-aws-sts-endpoints/

> Although the global (legacy) AWS STS endpoint https://sts.amazonaws.com is highly available, it’s hosted in a single AWS Region—US East (N. Virginia)—and like other endpoints, it doesn’t provide automatic failover to endpoints in other Regions. In this post I will show you how to use Regional AWS STS endpoints in your configurations to improve the performance and resiliency of your workloads.